packages/runtime/src/next/client.ts

```diff
@@ -201,23 +230,51 @@ export class Makeswift {
     return response
   }
 
-  async getPages({
-    siteVersion = MakeswiftSiteVersion.Live,
-  }: {
-    siteVersion?: MakeswiftSiteVersion
-  } = {}): Promise<MakeswiftPage[]> {
-    const response = await this.fetch(`v3/pages`, siteVersion)
+  private getPagesInternal = async (
+    args: {
+      siteVersion?: MakeswiftSiteVersion
+      limit?: number
+      after?: string
+      sortBy?: 'title' | 'path' | 'createdAt' | 'updatedAt' | 'publishedAt'
+      sortDirection?: 'asc' | 'desc'
+      includeOffline?: boolean
+      pathPrefix?: string
+      locale?: string
+    } = {},
+  ) => {
+    const {
+      siteVersion = MakeswiftSiteVersion.Live,
+      limit = 20,
+      after,
+      includeOffline,
+      sortBy,
+      sortDirection,
+      pathPrefix,
+      locale,
+    } = args
+
+    const url = new URL('v4/pages', this.apiOrigin)
+    url.searchParams.set('limit', limit.toString())
+    if (after != null) url.searchParams.set('after', after)
+    if (includeOffline != null) url.searchParams.set('includeOffline', String(includeOffline))
+    if (sortBy != null) url.searchParams.set('sortBy', sortBy)
+    if (sortDirection != null) url.searchParams.set('sortDirection', sortDirection)
+    if (pathPrefix != null) url.searchParams.set('pathPrefix', pathPrefix)
+    if (locale != null) url.searchParams.set('locale', locale)
+
+    const response = await this.fetch(url.pathname + url.search, siteVersion)
 
     if (!response.ok) {
       console.error('Failed to get pages', await response.json())
       throw new Error(`Failed to get pages with error: "${response.statusText}"`)
     }
 
-    const json = await response.json()
-
-    return json
+    const result = await response.json()
+    return makeswiftGetPagesResultAPISchema.parse(result)
   }
 
+  getPages = toIterablePaginationResult(this.getPagesInternal)
```

Comment:
Is it possible to have `getPages` be a method or do we have to settle for a class field if we're going to decorate it like this?

Reply:
It's possible to make it a method at the cost of giving the arguments record type a name. Does it matter if it's a method? I don't think we can make `toIterablePaginationResult` a decorator since it changes the signature.